src/main/java/mate/jdbc/manufacturerdao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,108 @@
+            while (resultSet.next()) {
+                String manufacturerName = resultSet.getString("name");
+                String manufacturerCountry = resultSet.getString("country");
+                Long id = resultSet.getObject("id", Long.class);
+                Manufacturer manufacturer =
+                        new Manufacturer(id, manufacturerName, manufacturerCountry);
+                manufacturers.add(manufacturer);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", throwables);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String sqlQuery =
+                "SELECT name, country FROM manufacturers WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(new Manufacturer(id, resultSet.getString("name"),
+                        resultSet.getString("country")));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Problem getting manufacturer from DB, id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String sqlQuery = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlQuery);) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer :(", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(deleteRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setLong(1, id);
+            return createManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't insert manufacturer to Data Base", throwables);
```

Comment:
```suggestion
        } catch (SQLException e) {
            throw new DataProcessingException("Can't delete manufacturer from DB", e);
```
be careful when copypasting